source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,177 @@
+
+In-App Calibration
+^^^^^^^^^^^^^^^^^^
+To calibrate your camera from a video file, click on :guilabel:`Calibrate Camera`
+
+.. image:: images/Calibrate.png
+    :alt: Calibrate
+
+There are two options for calibrating your camera, :guilabel:`OpenCV` and :guilabel:`MRcal`. It is generally recommended to calibrate with :guilabel:`MRcal`, as it has slightly better accuracy and is more robust against bad calibration data. The calibration option you use will dictate the type of board you calibrate with.
+
+:guilabel:`OpenCV` calibration uses a ChArUco board:
+
+.. image:: images/ChArUco.png
+    :alt: ChArUco
+
+:guilabel:`MRcal` uses a Checkerboard pattern:
+
+.. image:: images/Chessboard.png
+    :alt: Checkerboard
+
+.. note:: Regardless of which calibration option you use, make sure your calibration board has not been bent or creased, and is lying on a flat surface.
+.. note:: It does not matter if you use the exact same board as shown in the images, as there are fields in each calibration option to customize the calibration for your specific board.
+
+There are a couple of common fields that both :guilabel:`OpenCV` and :guilabel:`MRcal` calibration options use:
+
+* :guilabel:`Select Camera Calibration Video` is the button to select your video file.
+
+* :guilabel:`Square Width` is the width of the squares (in inches) of the board.
+
+* :guilabel:`Board Width` is the width (in squares) of the board
+
+* :guilabel:`Board Height` is the height (in squares) of the board
+
+For :guilabel:`OpenCV`, there is one more option to fill in:
+
+* :guilabel:`Marker Width` is the width of the ChArUco markers (in inches) on the board
+
+.. image:: images/OpenCV.png
+    :alt: OpenCV
+
+For :guilabel:`MRcal`, there are three more options to fill in:
+
+* :guilabel:`Image Width` is the width (in pixels) of the video's resolution
+* :guilabel:`Image Height` is the height (in pixels) of the video's resolution
+* :guilabel:`Focal Length` is the focal length (in pixels) of the camera.
+
+.. image:: images/MRcal.png
+    :alt: MRcal
+
+.. important:: For the calibration to work properly, it is important that you enter the exact width of the squares (and markers for ChArUco). It is recommended to use calipers to accurately measure these widths.
+
+When all the calibration fields have been entered, select :guilabel:`Calibrate` to run the calibration. A window will appear showing you the corner detection process:
+
+ChArUco:
+
+.. image:: images/ChArUcoDetection.png
+    :alt: ChArUcoDetection
+
+Checkerboard:
+
+.. image:: images/ChessboardDetection.png
+    :alt: CheckerboardDetection
+
+.. warning:: If any frames from the camera calibration look suspect, take a new video of the calibration board and try again.
+
+The camera calibration will automatically load the generated camera intrinsics JSON into WPIcal to continue to field calibration, but will also output the JSON file so it can be used for future calibrations.
```

Comment:
Is the JSON saved to a file with a standard location and file name? Or else how is it output?